core/src/main/java/org/apache/iceberg/rest/RESTCatalog.java

```diff
@@ -0,0 +1,417 @@
+import org.apache.iceberg.hadoop.Configurable;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.io.ResolvingFileIO;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.apache.iceberg.rest.requests.CreateNamespaceRequest;
+import org.apache.iceberg.rest.requests.CreateTableRequest;
+import org.apache.iceberg.rest.requests.UpdateNamespacePropertiesRequest;
+import org.apache.iceberg.rest.responses.CreateNamespaceResponse;
+import org.apache.iceberg.rest.responses.DropNamespaceResponse;
+import org.apache.iceberg.rest.responses.DropTableResponse;
+import org.apache.iceberg.rest.responses.GetNamespaceResponse;
+import org.apache.iceberg.rest.responses.ListNamespacesResponse;
+import org.apache.iceberg.rest.responses.ListTablesResponse;
+import org.apache.iceberg.rest.responses.LoadTableResponse;
+import org.apache.iceberg.rest.responses.UpdateNamespacePropertiesResponse;
+import org.apache.iceberg.util.Pair;
+
+public class RESTCatalog implements Catalog, SupportsNamespaces, Configurable<Configuration> {
+  private final Function<Map<String, String>, RESTClient> clientBuilder;
+  private RESTClient client = null;
+  private String catalogName = null;
+  private Map<String, String> properties = null;
+  private Object conf = null;
+  private FileIO io = null;
+
+  RESTCatalog(Function<Map<String, String>, RESTClient> clientBuilder) {
+    this.clientBuilder = clientBuilder;
+  }
+
+  @Override
+  public void initialize(String name, Map<String, String> props) {
+    this.client = clientBuilder.apply(props);
+    this.catalogName = name;
+    this.properties = props;
+    String ioImpl = props.get(CatalogProperties.FILE_IO_IMPL);
+    this.io = CatalogUtil.loadFileIO(ioImpl != null ? ioImpl : ResolvingFileIO.class.getName(), props, conf);
+  }
+
+  @Override
+  public void setConf(Configuration newConf) {
+    this.conf = newConf;
+  }
+
+  @Override
+  public String name() {
+    return catalogName;
+  }
+
+  @Override
+  public List<TableIdentifier> listTables(Namespace namespace) {
+    String ns = RESTUtil.urlEncode(namespace);
+    ListTablesResponse response = client
+        .get("v1/namespaces/" + ns + "/tables", ListTablesResponse.class, ErrorHandlers.namespaceErrorHandler());
+    return response.identifiers();
+  }
+
+  @Override
+  public boolean dropTable(TableIdentifier identifier, boolean purge) {
+    String tablePath = tablePath(identifier);
+    // TODO: support purge flag
+    DropTableResponse response = client.delete(
+        tablePath, DropTableResponse.class, ErrorHandlers.tableErrorHandler());
+    return response.isDropped();
+  }
+
+  @Override
+  public void renameTable(TableIdentifier from, TableIdentifier to) {
+
```

Comment:
Looks like this implementation may have been overlooked

Reply:
nevermind, just saw this wasn't slated for this PR

Reply:
Unsupported Operation Exception?